common/server_protocolhandler.cpp

```diff
@@ -542,6 +542,24 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
     if (!r)
         return Response::RespNameNotFound;
 
+    QString roomPermission = r->getRoomPermission().toLower();
+    if (roomPermission != "none"){
+        if (roomPermission == "registered") {
```

Comment:
I started to do it with the userlevel but noticed through out the existing code it uses this method of comparison and figured I wanted to stick with the style that was already present.